pkgs/by-name/pa/panda3ds/package.nix

```diff
@@ -0,0 +1,199 @@
+      luajit
+      libuv
+    ]
+    ++ lib.optionals withDiscordRPC [
+      discord-rpc
+      rapidjson
+    ]
+    ++ lib.optionals withVulkan [
+      vulkan-headers
+      vulkan-loader
+      glslang
+    ]
+    ++ lib.optionals withOpenGL [ libGL ]
+    ++ lib.optionals stdenv.hostPlatform.isLinux [
+      xorg.libX11
+      wayland
+    ];
+
+  # delete unused third_party deps
+  postUnpack = ''
+    rm -rf third_party/{SDL2,discord-rpc,fdk-aac,LuaJIT,boost}
+  '';
+
+  # fix missing ir/var/empty/constant_propagation_pass.cpp
+  postPatch = ''
+    mkdir -p third_party/dynarmic/src/dynarmic/ir/var
+    ln -s ../opt third_party/dynarmic/src/dynarmic/ir/var/empty
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    # main binary
+    install -Dm755 /build/source/build/Alber $out/bin/Alber
+
+    # desktop file
+    install -Dm444 $src/.github/Alber.desktop $out/share/applications/Alber.desktop
+
+    # icons
+    mkdir -p $out/share/icons/hicolor/256x256/apps
+    for size in 16 32 48 64 72 96 128 192 256 512 1024; do
+      mkdir -p $out/share/icons/hicolor/"$size"x"$size"/apps
+      magick convert \
+        -resize "$size"x"$size" \
+        -alpha on \
+        -background none \
+        $src/docs/img/Alber.png \
+        $out/share/icons/hicolor/"$size"x"$size"/apps/Alber.png
+    done
+
+    runHook postInstall
+  '';
+
+  postInstall =
+    let
+      libs = lib.makeLibraryPath (
+        lib.optionals withOpenGL [ libGL ]
+        ++ lib.optionals withVulkan [ vulkan-loader ]
+        ++ lib.optionals withDiscordRPC [ discord-rpc ]
+        ++ lib.optionals stdenv.hostPlatform.isLinux [ pulseaudio ]
+      );
+    in
+    if withQtGui then
+      ''
+        qtWrapperArgs+=(
+          --prefix LD_LIBRARY_PATH : ${libs}
+      ''
+      + (lib.optionalString (stdenv.hostPlatform.isLinux && !withQtWayland) ''
+        --set QT_QPA_PLATFORM "xcb"
+      '')
+      + ")"
+    else
+      ''
+        wrapProgram "$out/bin/Alber" \
+          --prefix LD_LIBRARY_PATH : ${libs}
+      '';
+
+  cmakeFlags = [
+    (lib.cmakeBool "ENABLE_USER_BUILD" true)
+    (lib.cmakeBool "DISABLE_PANIC_DEV" true)
+    (lib.cmakeBool "ENABLE_RENDERDOC_API" withRenderdoc)
+    (lib.cmakeBool "ENABLE_METAL" withMetal)
+    (lib.cmakeBool "ENABLE_VULKAN" withVulkan)
+    (lib.cmakeBool "ENABLE_OPENGL" withOpenGL)
+    (lib.cmakeBool "ENABLE_LUAJIT" withLuaJIT)
+    (lib.cmakeBool "ENABLE_DISCORD_RPC" withDiscordRPC)
+    (lib.cmakeBool "ENABLE_QT_GUI" withQtGui)
+    (lib.cmakeBool "DISABLE_SSE4" (!sse4Support))
+    (lib.cmakeBool "ENABLE_LTO" true)
+    (lib.cmakeBool "USE_SYSTEM_SDL2" true)
+    # "-DENABLE_TESTS=ON"
+  ];
+
+  meta = {
+    description = "HLE 3DS emulator" + lib.optionalString withQtGui " (Qt GUI)";
+    longDescription = ''
+      Panda3DS is an HLE, red-panda-themed Nintendo 3DS emulator written in C++ which started out as a fun project out
+       of curiosity, but evolved into something that can sort of play games!
```

Comment:
Judging by the sheer amount of overrides, I think it would be nice to document them here :p